src/main/java/com/attacktimer/VariableSpeed/Leagues.java

```diff
@@ -47,48 +47,67 @@ public int apply(final Client client, final AnimationData curAnimation, final At
         }
 
         AttackStyle attackStyle = Utils.getAttackStyle(client);
-
+        int masteryLevel;
         switch (leagueRelicVarbit)
         {
             case 0:
                 // No league relic active - player does not have t4 relic or is not in leagues.
                 return baseSpeed;
             case 1:
                 // Archer's Embrace (ranged).
+                // Ranged Mastery 3 & 5.
+                masteryLevel = client.getVarbitValue(Varbits.LEAGUES_RANGED_COMBAT_MASTERY_LEVEL);
                 if (attackStyle == AttackStyle.RANGING || attackStyle == AttackStyle.LONGRANGE)
                 {
-                    return applyRangedAndMeleeRelicSpeed(baseSpeed);
+                    return applyLeagueFormulaSpeed(baseSpeed, masteryLevel);
                 }
                 break;
             case 2:
                 // Brawler's Resolve (melee)
+                // Melee Mastery 3 & 5.
+                masteryLevel = client.getVarbitValue(Varbits.LEAGUES_MELEE_COMBAT_MASTERY_LEVEL);
                 if (attackStyle == AttackStyle.ACCURATE ||
                         attackStyle == AttackStyle.AGGRESSIVE ||
                         attackStyle == AttackStyle.CONTROLLED ||
                         attackStyle == AttackStyle.DEFENSIVE)
                 {
-                    return applyRangedAndMeleeRelicSpeed(baseSpeed);
+                    return applyLeagueFormulaSpeed(baseSpeed, masteryLevel);
                 }
                 break;
             case 3:
                 // Superior Sorcerer (magic)
+                // Magic Mastery 3 & 5.
+                masteryLevel = client.getVarbitValue(Varbits.LEAGUES_MAGIC_COMBAT_MASTERY_LEVEL);
                 if (attackStyle == AttackStyle.CASTING || attackStyle == AttackStyle.DEFENSIVE_CASTING)
                 {
-                    return 2;
+                    return applyLeagueFormulaSpeed(baseSpeed, masteryLevel);

```

Comment:
The current leagues doesn't use this leagueRelicVarbit. it should be sufficient to check the mastery level and attack style only.

Reply:
Makes sense, I'll remove it next time I'm at my PC.